### PR TITLE
Fixed a bug that raises `CablePath matching query does not exist` exception when deleting a device with multiple types of connected interfaces.

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes: #4416 
# What's Changed

Added `try...except..`. block to catch `Cable.DoesNotExist` error per discussion in standup.

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by glennmatthews — COMMENTED** [review]
Change looks good to me. Any chance of adding a unit test that would catch this issue?

File: nautobot/dcim/api/serializers.py [hunk 1]
```diff
@@ -173,25 +173,34 @@ class PathEndpointModelSerializerMixin(ValidatedModelSerializer):
 
     @extend_schema_field(serializers.CharField(allow_null=True))
     def get_connected_endpoint_type(self, obj):
-        if obj._path is not None and obj._path.destination is not None:
-            return f"{obj._path.destination._meta.app_label}.{obj._path.destination._meta.model_name}"
+        try:
+            if obj._path is not None and obj._path.destination is not None:
+                return f"{obj._path.destination._meta.app_label}.{obj._path.destination._meta.model_name}"
+        except CablePath.DoesNotExist:
+            pass
```
Comment (glennmatthews) [inline nautobot/dcim/api/serializers.py:180]: Consider (as @timizuoebideri1 has been doing in some of his recent code changes) using [contextlib.suppress](https://docs.python.org/3/library/contextlib.html#contextlib.suppress) for these, I think that would be for example:

```suggestion
        with contextlib.suppress(CablePath.DoesNotExist):
            if obj._path is not None and obj._path.destination is not None:
                return f"{obj._path.destination._meta.app_label}.{obj._path.destination._meta.model_name}"
```

File: nautobot/dcim/tests/test_models.py [hunk 2]
```diff
@@ -1423,6 +1429,61 @@ def test_create_cable_with_missing_status_connected(self):
 
         self.assertTrue(Cable.objects.filter(id=cable.pk).exists())
 
+    def test_deleting_device_with_multiple_types_of_connected_interfaces_successful(self):
```
Comment (glennmatthews) [inline nautobot/dcim/tests/test_models.py:1432]: Would be good to add a docstring noting that this is testing for #4416.

File: nautobot/dcim/tests/test_models.py [hunk 3]
```diff
@@ -1423,6 +1429,61 @@ def test_create_cable_with_missing_status_connected(self):
...
+        Cable.objects.all().delete()
+        interface1 = Interface.objects.create(device=self.device1, name="ethernet1")
+        interface2 = Interface.objects.create(device=self.device2, name="ethernet2")
+        interface3 = Interface.objects.create(device=self.device1, name="ethernet3")
+        interface4 = Interface.objects.create(device=self.device2, name="ethernet4")
```
Comment (glennmatthews) [inline nautobot/dcim/tests/test_models.py:1437]: Missing a `status` on these Interfaces. Can/should we use the `self.interfaceX` interfaces already created in `setUpTestData` instead of creating new ones?

File: nautobot/dcim/tests/test_models.py [hunk 4]
```diff
@@ -1423,6 +1429,61 @@ def test_create_cable_with_missing_status_connected(self):
 
         self.assertTrue(Cable.objects.filter(id=cable.pk).exists())
 
+    def test_deleting_device_with_multiple_types_of_connected_interfaces_successful(self):
+        Cable.objects.all().delete()
```
Comment (glennmatthews) [inline nautobot/dcim/tests/test_models.py:1433]: Why is this needed?

**Review by glennmatthews — COMMENTED** [review]
This initial PR should probably be targeted at `develop` rather than `ltm/1.6`; we can then make the call as to whether to backport/cherrypick the fix to the LTM branch after it's merged in `develop`.

File: nautobot/dcim/tests/test_models.py (hunk 4 above)
Comment (HanlinMiao) [reply nautobot/dcim/tests/test_models.py:1439]: This is more or less for me trying to start the test with a clean slate. It was necessary because I ran into duplicate cables when creating new cables.

File: nautobot/dcim/api/serializers.py [hunk 5]
```diff
@@ -145,17 +148,18 @@ def get_connected_endpoint(self, obj):
...
-        if obj._path is not None and obj._path.destination is not None:
-            depth = get_nested_serializer_depth(self)
-            return return_nested_serializer_data_based_on_depth(
-                self, depth, obj, obj._path.destination, "connected_endpoint"
-            )
+        with contextlib.suppress(CablePath.DoesNotExist):
+            if obj._path is not None and obj._path.destination is not None:
+                serializer = get_serializer_for_model(obj._path.destination, prefix="Nested")
+                context = {"request": self.context["request"]}
+                return serializer(obj._path.destination, context=context).data
```
Comment (glennmatthews) [inline nautobot/dcim/api/serializers.py:155]: Changes here look suspect - please double-check.

Comment (HanlinMiao) [reply nautobot/dcim/api/serializers.py]: Sorry for the oversight when rebasing, should be fixed now.

**Review by glennmatthews — APPROVED** [review]

**Comment by EdificomSA** [comment]
Thanks for the fix, I face the issue on 1.6.4, would it be possible to cherrypick/backport it ?

**Comment by milan2655** [comment]
Hallo!

I'm still facing this bug with the nautobot v1.6.22:
